### PR TITLE
Some fixes IDL compiler for parsing annotations

### PR DESCRIPTION
**Pull request by dennis-adlink**
- fix a bug in annotation_appl parameter handling
- typo fix in extensibility annotation enum value for appendable
- enable tests for parsing extensibility annotations

### CODE REVIEW

File: src/idl/src/scope.c [hunk 1]
```diff
@@ -452,7 +452,10 @@ idl_resolve(
   if (pstate->flags & IDL_FLAG_CASE_SENSITIVE)
     ignore_case = 0u;
 
-  scope = (scoped_name->absolute) ? pstate->global_scope : pstate->scope;
+  if (pstate->parser.state == IDL_PARSE_ANNOTATION_APPL_PARAMS)
+    scope = pstate->annotation_scope;
+  else
+    scope = (scoped_name->absolute) ? pstate->global_scope : pstate->scope;
```
Comment (k0ekk0ek) [inline src/idl/src/scope.c:458]: I'm not sure this is correct. It reads like the current scope isn't searched if something can't be found in the annotation declaration scope. I'm also not sure what the spec says about this... Should the scenario below work?

```
module m1 {
  @annotation a1 {
    unsigned int32 i1;
  };
  module m2 {
    const unsigned int32 i2 = 1;
    @a1(i1 = i2) struct s1 {
      char c1;
    };
  };
};
```

**Review by k0ekk0ek — CHANGES_REQUESTED** [review]
Thanks @dennis-adlink! I mostly agree, but there's a few minor things. Next to whether the proper scope is searched (we can ask @e-hndrks for this too), I also think that the name being looked up should be leading(?). If an absolute name is specified, we should start searching from the global scope and choose the annotation scope or current scope on fallback. Perhaps, we should also add some simple tests for those scenarios. Of course, since you're correcting my work I'm definitely willing to help out there :slightly_smiling_face:

File: src/idl/src/scope.c (hunk 1 above)
Comment (dennis-adlink) [reply src/idl/src/scope.c]: Good question, I'm not sure if it should and I can't really find the answer in the spec. The specs states that enums that are defined in the annotation body may be used unscoped, which could imply that a scoped name should be used for const expressions from outside the annotation body scope (i.e. `i1 = m2::i2') and that works with the changes I've added in commit 7b1f554. This commit also adds an additional test for annotation parameter scope.

File: src/idl/src/tree.c [hunk 2]
```diff
@@ -2493,7 +2493,7 @@ static void *iterate_annotation(const void *ptr, const void *cur)
   const idl_node_t *node = cur;
   assert(root);
   if (node) {
-    assert(idl_parent(root) == node);
+    assert(idl_parent(node) == root);
```
Comment (k0ekk0ek) [inline src/idl/src/tree.c:2496]: Ah, this probably saves me some work. Noticed a bug yesterday that may be caused by this :+1:

**Review by k0ekk0ek — APPROVED** [review]

**Comment by k0ekk0ek** [comment]
I approve, but it seems Clang-Tidy doesn't yet agree. If you fix that one, I'm happy to merge. Thanks @dennis-adlink!

**Pull request merged**